src/Aspire.Dashboard/Components/ResourcesGridColumns/ResourceNameDisplay.razor

```diff
@@ -6,18 +6,22 @@
 
 @inject IStringLocalizer<Columns> Loc
 
-<span class="resource-name-text" title="@FormatName(Resource)"><FluentHighlighter HighlightedText="@FilterText" Text="@FormatName(Resource)" /></span>
-
-@if (Resource.Properties.TryGetValue(KnownProperties.Container.Lifetime, out var value) &&
-    value.Value.HasStringValue &&
-    StringComparer.Ordinal.Equals(value.Value.StringValue, KnownContainerLifetimes.Persistent))
-{
-    <FluentIcon
-        Icon="Icons.Regular.Size16.Pin"
-        Title="@Loc[nameof(Columns.PersistentContainerIconTooltip)]"
-        Color="Color.Neutral"
-        Class="persistent-container-icon" />
-}
+<span class="resource-name-container">
+    <FluentIcon Width="16px" Class="resource-icon "
+        Value="@ResourceIconHelpers.GetIconForResource(Resource).WithColor(ColorGenerator.Instance.GetColorHexByKey(FormatName(Resource)))" />
```

Comment:
These are 24px icons displayed at 16px. It would be better if the 16px versions were used directly, with a fallback to shrinking a larger size icon if required. The 16px versions could have optimizations for smaller size.

Reply:
I think what this would look like is GetIconForResource takes size and returns the closest size we can to the one you asked for. Most of the icons we use here don't have 16px versions IIRC. So they would be a mix of 20/24px and you would need to know to request a size and also set width. I didn't do it this way because of that complexity in the API. The API would look like you could request a size and you would get what you asked for. if you assumed that you would be surprised. Assuming I'm not wrong about what you had in mind I will do this and try and make it as obvious as I can that you might not get the size you are requesting.

Reply:
Reverting my previous thought. I just checked in the fluentui-system-icons repo, and I do not think it's going to be reasonable to request an additional size for the several icons missing a 16px version.

Please use size 16 for the icons that support that size, and downscale the others